fairseq/sequence_generator.py

```diff
@@ -246,6 +247,12 @@ def _generate(
         scores = (
             torch.zeros(bsz * beam_size, max_len + 1).to(src_tokens).float()
         )  # +1 for eos; pad is never chosen for scoring
+
+        # initialize negative buffers
+        # help to select none ended candidates
+        negative_active_hypos = torch.arange(beam_size).repeat(bsz, 1)
+        negative_active_hypos = negative_active_hypos.to(src_tokens.device).long()
```

Comment:
This can be moved to `LexicallyConstrainedBeamSearch::step` and initialized lazily on the first step

Reply:
Done.